source/SIL.AppBuilder.Portal.Frontend/src/ui/components/header/notifications/display.tsx

```diff
@@ -1,91 +1,92 @@
+    if (visible) {
+      markAllAsViewed();
+      refetch();
+    }
+
+    toggleVisible();
   };
 
-  render() {
-    const { t } = this.props;
-
-    const {
-      notifications,
-      haveAllNotificationsBeenSeen,
-      isThereAtLeastOneNotificationToShow,
-    } = this.props;
-
-    const hasNotifications =
-      notifications && notifications.length > 0 && isThereAtLeastOneNotificationToShow;
-
-    const isMenuVisible = this.state.visible ? 'visible' : '';
-
-    return (
-      <div className='ui top right pointing dropdown' data-test-header-notification>
-        <div
-          style={{ zIndex: 1 }}
-          className={`full-overlay transition ${this.state.visible ? 'visible invisible' : ''}`}
-          onClick={this.toggle}
-        />
-
-        <div
-          data-test-notification-trigger
-          data-test-notification-active={!haveAllNotificationsBeenSeen}
-          style={{ position: 'relative' }}
-          onClick={this.toggle}
-        >
-          {haveAllNotificationsBeenSeen ? <NotificationIcon /> : <NotificationActiveIcon />}
-        </div>
-
-        <div className={`ui menu transition notifications ${isMenuVisible}`}>
-          {hasNotifications && (
-            <>
-              <div className='notification-buttons'>
-                <a href='#' data-test-clear-all onClick={this.clearAll}>
-                  {t('header.clearAll')}
-                </a>
-              </div>
-
-              <div className={notifications.length > 3 ? 'scrollable-menu' : ''}>
-                {notifications.map((notification) => (
-                  <Row key={notification.id} notification={notification} />
-                ))}
-              </div>
-            </>
-          )}
-
-          {!hasNotifications && (
-            <div className='notification-no-data'>{t('header.emptyNotifications')}</div>
-          )}
-        </div>
+  const isMenuVisible = visible ? 'visible' : '';
+
+  return (
+    <div className='ui top right pointing dropdown' data-test-header-notification>
+      <div
+        style={{ zIndex: 1 }}
+        className={`full-overlay transition ${visible ? 'visible invisible' : ''}`}
+        onClick={toggle}
+      />
+
+      <div
+        data-test-notification-trigger
+        data-test-notification-active={!haveAllNotificationsBeenSeen}
+        style={{ position: 'relative' }}
+        onClick={toggle}
+      >
+        {haveAllNotificationsBeenSeen ? <NotificationIcon /> : <NotificationActiveIcon />}
       </div>
-    );
-  }
-}
 
-export default Notifications;
+      <div className={`ui menu transition notifications ${isMenuVisible}`}>
+        {hasNotifications && (
+          <>
+            <div className='notification-buttons'>
+              <a href='#' data-test-clear-all onClick={preventDefault(clearAll)}>
+                {t('header.clearAll')}
+              </a>
+            </div>
+
+            <div className={notifications.length > 3 ? 'scrollable-menu' : ''}>
```

Comment:
what is scrollable-menu doing? Why wouldn't we just apply it always and let the browser overflow if necessary?

Reply:
totally agree. I didn't implement this logic, I just refactored the component from a class to function